### PR TITLE
Port buffer-enable-undo

### LINKED ISSUE #1155: Port buffer-enable-undo
```rust
DEFUN ("buffer-enable-undo", Fbuffer_enable_undo, Sbuffer_enable_undo,
       0, 1, "",
       doc: /* Start keeping undo information for buffer BUFFER.
No argument or nil as argument means do this for the current buffer.  */)
  (register Lisp_Object buffer)
{
  Lisp_Object real_buffer;

  if (NILP (buffer))
    XSETBUFFER (real_buffer, current_buffer);
  else
    {
      real_buffer = Fget_buffer (buffer);
      if (NILP (real_buffer))
	nsberror (buffer);
    }

  if (EQ (BVAR (XBUFFER (real_buffer), undo_list), Qt))
    bset_undo_list (XBUFFER (real_buffer), Qnil);

  return Qnil;
}
```

### PR DESCRIPTION
**Pull request by odontomachus**
Fixes #1155.

### CODE REVIEW

File: rust_src/src/buffers.rs [hunk 1]
```diff
@@ -924,6 +924,18 @@ impl From<LispBufferOrCurrent> for LispBufferRef {
...
+/// Return false.
+/// If the optional arg BUFFER is provided and not nil, enable undoes in that
+/// buffer, otherwise run on the current buffer.
+#[lisp_fn(min = "0")]
+pub fn buffer_enable_undo(buffer: LispBufferOrCurrent) -> LispObject {
+    let mut buf: LispBufferRef = buffer.into();
+    if buf.undo_list_ == Qt {
+        buf.undo_list_ = Qnil;
+    }
+    Qnil
```
Comment (shaleh) [inline rust_src/src/buffers.rs]: When a DEFUN unconditionally returns `Qnil` the Rust version can be defined without a return value. Equivalent to `void func(....) { }` in C.

File: rust_src/src/buffers.rs [hunk 2]
```diff
@@ -924,6 +924,18 @@ impl From<LispBufferOrCurrent> for LispBufferRef {
     }
 }
 
+/// Return false.
+/// If the optional arg BUFFER is provided and not nil, enable undoes in that
+/// buffer, otherwise run on the current buffer.
+#[lisp_fn(min = "0")]
+pub fn buffer_enable_undo(buffer: LispBufferOrCurrent) -> LispObject {
+    let mut buf: LispBufferRef = buffer.into();
+    if buf.undo_list_ == Qt {
```
Comment (shaleh) [inline rust_src/src/buffers.rs]: We typically write this  `obj.eq(Qt)`

File: rust_src/src/buffers.rs [hunk 3]
```diff
@@ -924,6 +924,18 @@ impl From<LispBufferOrCurrent> for LispBufferRef {
     }
 }
 
+/// Return false.
+/// If the optional arg BUFFER is provided and not nil, enable undoes in that
+/// buffer, otherwise run on the current buffer.
+#[lisp_fn(min = "0")]
```
Comment (brotzeit) [inline rust_src/src/buffers.rs]: We need to add `intspec = ""` here.

**Comment by odontomachus** [comment]
I've made the changes. Thanks for the feedback.

File: rust_src/src/buffers.rs (hunk 3 above)
Comment (shaleh) [reply rust_src/src/buffers.rs]: I missed that. Thanks.

@odontomachus there is a tool in rust_src/admin called `rustize-emacs` which auto mates the creation of the Rust lisp_fn macro and the Rust function. It handles the interactive check for you as well as the min= if needed.

**Pull request merged**